tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -637,6 +635,7 @@ TEST_F(ValidationTest, WhenDepthNotFloatThenFail) {
 }
 
 TEST_F(ValidationTest, BarrierFail) {
+  if (!m_CompilerPreservesBBNames) return;    // This test requires name preservation to succeed currently.
```

Comment:
Is it possible to mark the test as 'Blocked' like we do for execution tests instead of silently passing?